packages/blocks/src/text-input/index.js

```diff
@@ -49,12 +49,13 @@ const TextInput = ( { attributes, className } ) => {
 				<RawHTML>{ attributes.label }</RawHTML>
 			</FormInputWrapper.Label>
 			<FormTextInput
+				value={ fieldValue }
+				placeholder={ attributes.placeholder }
+				onChange={ ( event ) => onUpdate( event.target.value ) }
```

Comment:
We should do this inside `<FormTextInput />`, so here you just write `onChange={ onUpdate }` or `onChange={ setFieldValue }`.